doc/source/whatsnew/v0.17.0.txt

```diff
@@ -330,6 +333,47 @@ Other API Changes
   ``raise ValueError``                All other public methods (names not beginning with underscores)
   ===============================     ===============================================================
 
+
+- default behavior for HDF write functions with "table" format is now to keep rows that are all missing except for index. (:issue:`9382`)
+
+Previously,
+
```

Comment:
create and show ``df_with_missing`` in an ``ipython`` block here (and remove from below), so we can see what it is.

Further I would have also a row with a single ``nan`` (to illustrate that it is unchanged)